mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -231,4 +242,57 @@ public void searchShouldNotReturnResultsFromOtherUsers() throws Exception {
         assertThat(metaDatas.get(0).getPath()).isEqualTo(MailboxPath.inbox(session));
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldUpdateStoredAnnotation() throws MailboxException {
+        MailboxPath inbox = setUpMailbox();
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(PRIVATE_ANNOTATION));
+
+        mailboxManager.updateAnnotations(inbox, session, Lists.newArrayList(PRIVATE_ANNOTATION_UPDATE));
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEqualTo(ImmutableList.of(PRIVATE_ANNOTATION_UPDATE));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldDeleteAnnotationWithNilValue() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(PRIVATE_ANNOTATION));
+
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(MailboxAnnotation.nil("/private/comment")));
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEmpty();
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldReturnEmptyForNonStoredAnnotation() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEmpty();
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldRetrieveStoredAnnotations() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        mailboxManager.updateAnnotations(inbox, session, ANNOTATIONS);
+
+        assertThat(mailboxManager.getAllAnnotations(inbox, session)).isEqualTo(ANNOTATIONS);
+    }
+
+    @ContractTest
+    public void getAnnotationsShouldRetrieveStoresAnnotationsByKeys() throws BadCredentialsException, MailboxException {
+        MailboxPath inbox = setUpMailbox();
+
+        mailboxManager.updateAnnotations(inbox, session, ANNOTATIONS);
+
+        assertThat(mailboxManager.getAnnotationsByKeys(inbox, session, ImmutableList.of("/private/comment")))
```

Comment:
assertj has assertions for almost any type, here you could use `.containsExactly(PRIVATE_ANNOTATION)`